activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -1032,7 +1032,9 @@ def build_statement_pool
 
         def can_perform_case_insensitive_comparison_for?(column)
           @case_insensitive_cache ||= {}
-          @case_insensitive_cache[column.sql_type] ||= begin
+          return @case_insensitive_cache[column.sql_type] if @case_insensitive_cache.key?(column.sql_type)
+
+          @case_insensitive_cache[column.sql_type] = begin
```

Comment:
A bit of a nitpick, but I think using `Hash#fetch` would be cleaner, and would save one lookup on the fastpath, something like:

```ruby
@case_insensitive_cache.fetch(column.sql_type) do |key|
  @case_insensitive_cache[key] = begin
```